src/goto-instrument/dump_c.cpp

```diff
@@ -1432,3 +1496,56 @@ void dump_cpp(
     new_cpp_language);
   out << goto2cpp;
 }
+
+static bool
+module_local_declaration(const symbolt &symbol, const std::string module)
+{
+  std::string base_name =
+    get_base_name(id2string(symbol.location.get_file()), true);
+  std::string symbol_module = strip_string(id2string(symbol.module));
+  if(base_name == module && symbol_module == module)
+  {
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
+
+void dump_c_type_header(
+  const goto_functionst &src,
+  const bool use_system_headers,
+  const bool use_all_headers,
+  const bool include_harness,
+  const namespacet &ns,
+  const std::string module,
+  std::ostream &out)
+{
+  symbol_tablet symbol_table = ns.get_symbol_table();
+  for(symbol_tablet::iteratort it = symbol_table.begin();
+      it != symbol_table.end();
+      it++)
```

Comment:
range for?

Reply:
I am not aware of telling the ranged for loop how to pick this iterator.
`for(auto it : symbol_table){}` iterates a slightly different data structure.

Reply:
Strange, it should pick up `.begin()` and `.end()`. At least you should use ++it instead of it++.